system-tests/azure-blob-transfer-fixtures/src/testFixtures/java/org/eclipse/edc/test/system/blob/BlobTransferValidator.java

```diff
@@ -26,11 +28,13 @@ public class BlobTransferValidator implements ThrowingConsumer<Map<String, Objec
     private final BlobServiceClient client;
     private final String expectedContent;
     private final String expectedName;
+    private final Map<String, String> expectedMetadata;
 
-    public BlobTransferValidator(BlobServiceClient client, String expectedContent, String expectedName) {
+    public BlobTransferValidator(BlobServiceClient client, String expectedContent, String expectedName, @Nullable Map<String, String> expectedMetadata) {
```

Comment:
nit: I'd prefer having an empty map instead of a nullable one, simply because then we don't have to check for nullity later any special treatment